pkg/api/v1/atlasdatabaseuser_types_test.go

```diff
@@ -0,0 +1,157 @@
+package v1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/cel"
+)
+
+func TestProjectReference(t *testing.T) {
+	validator, err := cel.VersionValidatorFromFile(t, "../../../config/crd/bases/atlas.mongodb.com_atlasdatabaseusers.yaml", "v1")
+	require.NoError(t, err)
+
+	tests := map[string]struct {
+		dbUser         *AtlasDatabaseUser
+		expectedErrors []string
+	}{
+		"no project reference is set": {
+			dbUser: &AtlasDatabaseUser{
+				Spec: AtlasDatabaseUserSpec{},
+			},
+			expectedErrors: []string{"spec: Invalid value: \"object\": must define only one project reference through externalProjectRef or projectRef"},
+		},
+		"both project references are set": {
+			dbUser: &AtlasDatabaseUser{
+				Spec: AtlasDatabaseUserSpec{
+					Project: &common.ResourceRefNamespaced{
+						Name: "my-project",
+					},
+					ExternalProjectRef: &ExternalProjectReference{
+						ID: "my-project-id",
+					},
+				},
+			},
+			expectedErrors: []string{
+				"spec: Invalid value: \"object\": must define only one project reference through externalProjectRef or projectRef",
+				"spec: Invalid value: \"object\": must define a local connection secret when referencing an external project",
+			},
+		},
+		"external project references is set": {
+			dbUser: &AtlasDatabaseUser{
+				Spec: AtlasDatabaseUserSpec{
+					ExternalProjectRef: &ExternalProjectReference{
```

Comment:
let's add a test case where we have an external project ref and a local connection secret.